azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -384,21 +385,39 @@ def send_output_event(self, message, callback=None):
     def _on_shared_access_string_updated(self):
         self._trig_on_shared_access_string_updated()
 
-    def enable_input_messages(self, callback=None, qos=1):
+    def enable_feature(self, feature_name, callback=None, qos=1):
+        if feature_name == constant.INPUT:
+            self._enable_input_messages(callback, qos)
+        elif feature_name == constant.C2D:
+            self._enable_c2d_messages(callback, qos)
+        else:
+            logger.error("Feature name {} is unknown".format(feature_name))
+            raise ValueError("Invalid feature name")
+
+    def disable_feature(self, feature_name, callback=None):
+        if feature_name == constant.INPUT:
+            self._disable_input_messages(callback)
```

Comment:
>_disable_input_messages [](start = 17, length = 23)

this big if/else statement may not grow well. fine for now, but if we get past 3 or 4 features that have enable/disable functions, we may want to establish a pattern here (dictionary or otherwise)